src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs

```diff
@@ -954,6 +954,79 @@ public static void Main()
 9");
         }
 
+        [Fact]
+        public void TypeCheckInPropertyPattern()
+        {
+            var source =
+@"using System;
+
+class Program2
+{
+    public static void Main()
+    {
+        object o = new Frog(1, 2);
+        if (o is Frog(1, 2))
+        {
+            Console.Write(1);
+        }
+        if (o is Frog { A: 1, B: 2 })
+        {
+            Console.Write(2);
+        }
+        if (o is Frog(1, 2) { A: 1, B: 2, C: 3 })
+        {
+            Console.Write(3);
+        }
```

Comment:
>} [](start = 8, length = 1)

Consider adding some `is` pattern conditions that should fail.

Reply:
Will do in a new iteration.

---
In reply to: [176463956](https://github.com/dotnet/roslyn/pull/25544#discussion_r176463956) [](ancestors = 176463956)